nextcord/automod.py

```diff
@@ -0,0 +1,333 @@
+
+    @property
+    def exempt_channels(self) -> List[GuildChannel]:
+        """List[:class:`GuildChannel`]: A list of channels that will not be affected by this rule. `[]` if not set."""
+        return [self.guild.get_channel(exempt_channel_id) for exempt_channel_id in self.exempt_channel_ids]  # type: ignore -- same
+
+    @overload
+    async def edit(
+        self,
+        *,
+        name: Optional[str] = ...,
+        event_type: Optional[EventType] = ...,
+        keyword_filters: Optional[List[str]] = ...,
+        notify_channel: Optional[GuildChannel] = ...,
+        timeout_seconds: Optional[int] = ...,
+        enabled: Optional[bool] = ...,
+        exempt_roles: Optional[List[Role]] = ...,
+        exempt_channels: Optional[List[GuildChannel]] = ...,
+    ):
+        ...
+
+    @overload
+    async def edit(
+        self,
+        *,
+        name: Optional[str] = ...,
+        event_type: Optional[EventType] = ...,
+        preset: Optional[KeywordPresetType] = ...,
+        notify_channel: Optional[GuildChannel] = ...,
+        timeout_seconds: Optional[int] = ...,
+        enabled: Optional[bool] = ...,
+        exempt_roles: Optional[List[Role]] = ...,
+        exempt_channels: Optional[List[GuildChannel]] = ...,
+    ):
+        ...
+
+    async def edit(self, **fields):
+        """
+        |coro|
+        Edit this auto moderation rule.
+
+        You must have the :attr:`~Permissions.manage_guild` to be able to do this.
+
+        Parameters
+        -----------
+        name: Optional[:class:`str`]
+            The new name of this auto moderation rule.
+        event_type: Optional[:class:`EventType`]
+            The new trigger event type of this auto moderation rule.
+        keyword_filters: Optional[List[:class:`str`]]
+            The keywords that the filter should match.
+
+            .. note::
+                This will only work if the rule's ``trigger_type`` is :attr:`TriggerType.keyword`.
+        presets: Optional[List[:class:`KeywordPresetType`]]
+            The keyword presets that the filter should match.
+
+            .. note::
+                This will only work if the rule's ``trigger_type`` is :attr:`TriggerType.keyword_preset`
+        notify_channel: Optional[:class:`abc.GuildChannel`]
+            The channel that will receive the notification when this rule is triggered. Cannot be mixed with ``notify_channels``.
+        timeout_seconds: Optional[:class:`int`]
+            The seconds to timeout the person triggered this rule.
+        enabled: Optional[:class:`bool`]
+            Whether if this rule is enabled.
+        exempt_roles: Optional[List[:class:`Role`]]
+            A list of roles that should not be affected by this rule.
+        exempt_channels: Optional[List[:class:`abc.GuildChannel`]]
+            A list of channels that should not be affected by this rule.
+        """
+        payload = {}
```

Comment:
there should be some check (like `embed` and `embeds` in `nextcord.abc.Messageable.send` against `keyword_filters` and `preset` as your overloads seems to show theyre mutually exclusive

Reply:
well theres tewo checks against two different trigger type so....

Reply:
*two

Reply:
a) You can edit comments
b) Yet there are no checks for mutual exclusivity explicitly? Your current checks imply to the end user that they just passed an argument wrong, rather then implying its unsupported. So they will go search for a solution to a non-existent problem

Reply:
done